src/parents.js

```diff
@@ -28,13 +34,44 @@ export function setAll (node, options ) {
  * @param {object} parent
  * @param {object} [options]
  * @param {boolean} [options.force] Allow overwriting
+ * @throws {Error} If the child node is not found in the parent
  */
 export function set (node, parent, { force } = {}) {
 	if (!force && parentMap.has(node)) {
 		// We assume that if the node already has a parent, its subtree will also have parents
 		return false;
 	}
-	parentMap.set(node, parent);
+
+	if (parent === undefined) {
+		parentMap.delete(node);
+		return;
+	}
+
+	if (parent === null) {
+		parentMap.set(node, {parent});
```

Comment:
I had to do a double take here, I wonder if something like this would make it clearer:
```suggestion
		// Node is the AST root
		parentMap.set(node, { parent: null });
```